002_simulation_structure/002_simulation_structure.tex

```diff
@@ -1,14 +1,108 @@
+    \label{fig:health_system}
+\end{figure}
+
+\subsection{Food and Health: \lstinline$updateHP$}\label{updateHP}
+To increase their HP, agents need to eat. However, the amount an agent's HP improves can saturate in a single day; eating more than a certain amount will provide an agent with no extra benefit to their HP. Moreover, eating more food will lead to diminishing returns in terms of HP change. Mathematically, the ideas of diminishing returns and saturation are well captured by the step response of a 1st-order system:
+
+\begin{equation}\label{updateHP_general}
+   \text{newHP}= \text{currentHP} +\underbrace{w(1-e^{\frac{-\text{foodTaken}}{\tau}})}_{\text{HPChange}}
+\end{equation}
+
+where the two parameters $w$ and $\tau$ are defined at the beginning of the simulation. The shape of this curve is given in \Cref{fig:updateHP} together with some important parameters.
+
+\begin{figure}[htb]%
+    \centering
+    \subfloat[\centering Overview]{{\includegraphics[width=0.36\linewidth]{002_simulation_structure/images/health_updateHP_overview.pdf}}}%
+    \qquad
+    \subfloat[\centering Detailed representation]{{\includegraphics[width=0.36\linewidth]{002_simulation_structure/images/health_updateHP_detailed.pdf}}}%
+    \caption{\texttt{updateHP} as a function of the amount of food eaten (``FoodTaken'').}%
+    \label{fig:updateHP}%
+\end{figure}
+
+It is not possible to gain more HP than $w$ over the duration of one day; this is an intentional limit to prevent an agent's health from improving too quickly. As an example, we can think of an agent that starts from the weak level and wants to reach the maximum HP value. It would take several days for this agent to ``recover'' from this weak level and stabilise its health to a high HP value. 
+
+Note that it is possible for an agent to achieve an HP value that is larger than \texttt{MaxHP} inside \lstinline$hpDecay()$. At the end of each day, the \lstinline$hpDecay()$ function will apply the cost of living and then bound the final HP value by \lstinline$MaxHP$.
+
+
+Agents in the critical state are treated differently. For these agents, HP is updated according to this equation:
+
+\begin{equation}\label{updateHP_critical}
+    \text{newHP} = \min\left\{\text{HPCritical}+\text{HPReqCToW}, \text{currentHP} +w(1-e^{\frac{-\text{foodTaken}}{\tau}})\right\}
+\end{equation}
+
+\subsection{Cost of Living: \lstinline$hpDecay()$}\label{hpDecay}
+At the end of each day, the HP value of the agents will be reduced by the cost of living. The cost of living is larger for an agent with larger HP value than for an agent with lower HP value. This fact is motivated by a simple observation: humans that have stronger bodies and immune systems also need more food to sustain their level of health. The exact relation between HP value, cost of living, and HP value after applying the cost of living is given by the following linear relation:
```

Comment:
see above comments about referring to equations in text. Change last line to: "The exact relation between HP value, cost of living, and HP value after applying the cost of living is given by the linear relation (\ref{hpDecay_equation})"

Reply:
done